dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/VertexAI/VertexAIGeminiEndpointProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+
+namespace Microsoft.SemanticKernel.Connectors.GoogleVertexAI;
+
+/// <summary>
+/// Provides a collection of endpoints for the Gemini Vertex AI API.
+/// </summary>
+internal sealed class VertexAIGeminiEndpointProvider : IEndpointProvider
+{
+    private readonly VertexAIConfiguration _configuration;
+
+    /// <summary>
+    /// Initializes a new instance of the GeminiEndpoints class with the specified API key.
+    /// </summary>
+    /// <param name="configuration">Vertex AI configuration.</param>
+    public VertexAIGeminiEndpointProvider(VertexAIConfiguration configuration)
+    {
+        this._configuration = configuration;
+    }
+
+    /// <summary>
+    /// Gets the base endpoint for the Gemini API.
+    /// </summary>
+    public Uri BaseEndpoint => new($"https://{this._configuration.Location}-aiplatform.googleapis.com/v1/projects/{this._configuration.ProjectId}/locations/{this._configuration.Location}/publishers/google/");
+
+    /// <summary>
+    /// Gets the endpoint URI for accessing the models in the Gemini API.
+    /// </summary>
+    public Uri ModelsEndpoint => new(this.BaseEndpoint, "models/");
+
+    /// <inheritdoc />
+    public Uri GetTextGenerationEndpoint(string modelId)
```

Comment:
nit.

If the informations regarding the endpoints is always the same, we can move that information to the property and not creating new string instances everytime the method is invoked. Not a big deal, but something to be considered.

```suggestion
    public Uri GetTextGenerationEndpoint(string modelId)
```

Reply:
I just did some small optimalization in this class, but I prefer leave passing modelId and creating new Uri Instance as is now. It's more convenient than passing modelId and EmbeddingModelId to EndpointProvider contructor.